### PR TITLE
Add cautionary note regarding floats as dotted keys

### LINKED ISSUE #616: Add an example about dotted numbers (3.14) as a dotted key
> Add a small note to the dotted keys section that `1.0 = true` gets parsed into `{"1": {"0": true"}}`.

_Originally posted by @pradyunsg in https://github.com/toml-lang/toml/issues/592#issuecomment-489573215_

### PR DESCRIPTION
**Pull request by pradyunsg**
Closes #616 
Closes #624

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -201,6 +201,20 @@ name = "Tom"
 name = "Pradyun"
 ```
 
+Since bare keys are allowed to compose of only ASCII integers, it is possible
+to write dotted keys that look like floats but are 2-part dotted keys. Don't do
+this unless you have a good reason to (you probably don't).
+
+```toml
+3.14159 = "pi"
```
Comment (yjqiang) [inline README.md]: How about add `"3.141592" = "pi"`?  By contrast, we can make it clearer.

Comment (pradyunsg) [reply README.md]: I wanted to make it clear what value came from what key. :)

Comment (yjqiang) [reply README.md]: 👌

**Comment by pradyunsg** [comment]
Thanks for taking a look @yjqiang! :D

**Pull request merged**